[
    {
        "Question_title":"What are the pros and cons of using DVC and Pachyderm?",
        "Question_body":"<p>What are the pros and cons of using either of these?<\/p>\n\n<p><a href=\"https:\/\/github.com\/iterative\/dvc\" rel=\"nofollow noreferrer\">https:\/\/github.com\/iterative\/dvc<\/a><\/p>\n\n<p><a href=\"https:\/\/github.com\/pachyderm\/pachyderm\" rel=\"nofollow noreferrer\">https:\/\/github.com\/pachyderm\/pachyderm<\/a><\/p>",
        "Question_answer_count":1,
        "Question_comment_count":1,
        "Question_creation_date":"2019-07-04 06:12:59.043 UTC",
        "Question_favorite_count":null,
        "Question_score":1,
        "Question_tags":"machine-learning|version-control|data-science|dvc|pachyderm",
        "Question_view_count":1635,
        "Owner_creation_date":"2019-07-04 06:06:43.123 UTC",
        "Owner_last_access_date":"2019-07-18 00:21:59.09 UTC",
        "Owner_location":null,
        "Owner_reputation":41,
        "Owner_up_votes":0,
        "Owner_down_votes":0,
        "Owner_views":3,
        "Answer_body":null,
        "Answer_comment_count":null,
        "Answer_creation_date":null,
        "Answer_last_edit_date":null,
        "Answer_score":null,
        "Question_last_edit_date":null,
        "Question_exclusive_tag":"Pachyderm"
    },
    {
        "Question_title":"Pachyderm pipeline unable to pull docker image",
        "Question_body":"<p>I have successfully pushed my docker image to the registry:\n<code>my-username\/image-name<\/code><\/p>\n\n<p>However, pachyderm is still unable to pull docker image(from logs):<\/p>\n\n<pre><code>container \"user\" in pod \"pipeline-mopng-beneficiary-v2-v1-b6kln\" is waiting to start: trying and failing to pull image\n<\/code><\/pre>\n\n<p>Where should I specify the image so that pachyderm is able to pull one?\nThis is my config file:<\/p>\n\n<pre><code>{\n    \"pipeline\": {\n        \"name\": \"mopng-beneficiary-v2\"\n    },\n    \"input\": {\n        \"pfs\": {\n            \"repo\": \"mopng_beneficiary_v2\",\n            \"glob\": \"\/*\"\n        }\n    },\n    \"transform\": {\n        \"cmd\": [\"python3\", \"\/pclean.py \/pfs\/mopng_beneficiary_v2\/euoEQHIwIQTe1wXtg46fFYok.csv \/pfs\/mopng_beneficiary_v2\/\/Users\/aviralsrivastava\/Downloads\/5Feb18_master_ujjwala_latlong_dist_dno_so_v7.csv \/pfs\/mopng_beneficiary_v2\/\/Users\/aviralsrivastava\/Downloads\/ppac_master_v3_mmi_enriched_with_sanity_check.csv \/pfs\/mopng_beneficiary_v2\/Qc.csv\"],\n        \"image\": \"username\/my-image\"\n    }\n}\n<\/code><\/pre>",
        "Question_answer_count":1,
        "Question_comment_count":1,
        "Question_creation_date":"2018-12-31 15:38:12.627 UTC",
        "Question_favorite_count":1.0,
        "Question_score":1,
        "Question_tags":"docker|pachyderm",
        "Question_view_count":158,
        "Owner_creation_date":"2018-07-10 15:06:30.283 UTC",
        "Owner_last_access_date":"2019-02-13 09:38:09.97 UTC",
        "Owner_location":"delhi",
        "Owner_reputation":903,
        "Owner_up_votes":53,
        "Owner_down_votes":0,
        "Owner_views":144,
        "Answer_body":null,
        "Answer_comment_count":null,
        "Answer_creation_date":null,
        "Answer_last_edit_date":null,
        "Answer_score":null,
        "Question_last_edit_date":null,
        "Question_exclusive_tag":"Pachyderm"
    },
    {
        "Question_title":"File not found even after adding the file inside docker",
        "Question_body":"<p>I have written a docker file which adds my python script inside the container:\n<code>ADD test_pclean.py \/test_pclean.py<\/code><\/p>\n\n<p>My directory structure is:<\/p>\n\n<pre><code>.\n\u251c\u2500\u2500 Dockerfile\n\u251c\u2500\u2500 README.md\n\u251c\u2500\u2500 pipeline.json\n\u2514\u2500\u2500 test_pclean.py\n<\/code><\/pre>\n\n<p>My json file which acts as a configuration file for creating a pipeline in Pachyderm is as follows:<\/p>\n\n<pre><code>{\n    \"pipeline\": {\n        \"name\": \"mopng-beneficiary-v2\"\n    },\n    \"transform\": {\n        \"cmd\": [\"python3\", \"\/test_pclean.py\"],\n        \"image\": \"avisrivastava254084\/mopng-beneficiary-v2-image-7\"\n    },\n    \"input\": {\n        \"atom\": {\n            \"repo\": \"mopng_beneficiary_v2\",\n            \"glob\": \"\/*\"\n        }\n    }\n}\n<\/code><\/pre>\n\n<p>Even though I have copied the official documentation's example, I am facing an error:\n<code>python3: can't open file '\/test_pclean.py': [Errno 2] No such file or directory<\/code><\/p>\n\n<p>My dockerfile is:<\/p>\n\n<pre><code>FROM    debian:stretch\n\n# Install opencv and matplotlib.\nRUN apt-get update \\\n    &amp;&amp; apt-get upgrade -y \\\n    &amp;&amp; apt-get install -y unzip wget build-essential \\\n        cmake git pkg-config libswscale-dev \\\n    &amp;&amp; apt-get clean \\\n    &amp;&amp; rm -rf \/var\/lib\/apt\n\nRUN apt update\nRUN apt-get -y install python3-pip\nRUN pip3 install matplotlib\nRUN pip3 install pandas\n\nADD test_pclean.py \/test_pclean.py\nENTRYPOINT [ \"\/bin\/bash\/\" ]\n<\/code><\/pre>",
        "Question_answer_count":2,
        "Question_comment_count":6,
        "Question_creation_date":"2018-12-31 22:48:53.713 UTC",
        "Question_favorite_count":null,
        "Question_score":0,
        "Question_tags":"python|python-3.x|docker|pachyderm",
        "Question_view_count":430,
        "Owner_creation_date":"2018-07-10 15:06:30.283 UTC",
        "Owner_last_access_date":"2019-02-13 09:38:09.97 UTC",
        "Owner_location":"delhi",
        "Owner_reputation":903,
        "Owner_up_votes":53,
        "Owner_down_votes":0,
        "Owner_views":144,
        "Answer_body":"<p>I was not changing the commits to my docker images on each build and hence, Kubernetes was using the local docker file that it had(w\/o tags and commits, it doesn't acknowledge any change). Once I started using commit with each build, Kubernetes started downloading the intended docker image.<\/p>",
        "Answer_comment_count":0.0,
        "Answer_creation_date":"2019-01-17 10:20:58.433 UTC",
        "Answer_last_edit_date":null,
        "Answer_score":0.0,
        "Question_last_edit_date":"2019-01-02 10:06:19.99 UTC",
        "Question_exclusive_tag":"Pachyderm"
    },
    {
        "Question_title":"How do you link Pachyderm with the correct Kubernetes context?",
        "Question_body":"<p>I have more than one Kubernetes context. When I change contexts, I have been using <code>kill -9<\/code>  to kill the port-forward in order to redo the <code>pachtctl port-forward &amp;<\/code> command. I wonder if this is the right way of doing it.<\/p>\n\n<p>In more detail:<\/p>\n\n<p>I start off being in a Kubernetes context, we'll call it context_x. I then want to change context to my local context, called minikube. I also want to see my repos for this minikube context, but when I use <code>pachctl list-repo<\/code>, it still shows context_x's Pachyderm repos. When I do <code>pachctl port-forward<\/code>, I then get an error message about the address being already in use. So I have to ps -a, then kill -9 on those port forward processes, and then do pachctl port-forward command again.<\/p>\n\n<p>An example of what I've been doing:<\/p>\n\n<pre><code>$ kubectl config use-context minikube\n$ pachctl list-repo #doesn't show minikube context's repos\n$ pachctl port-forward &amp;\n...several error messages along the lines of:\nUnable to create listener: Error listen tcp4 127.0.0.1:30650: bind: address already in use\n$ ps -a | grep forward\n33964 ttys002    0:00.51 kubectl port-forward dash-12345678-abcde 38080:8080\n33965 ttys002    0:00.51 kubectl port-forward dash-12345679-abcde 38081:8081\n37245 ttys002    0:00.12 pachctl port-forward &amp;\n37260 ttys002    0:00.20 kubectl port-forward pachd-4212312322-abcde 30650:650\n$ kill -9 37260\n$ pachctl port-forward &amp; #works as expected now\n<\/code><\/pre>\n\n<p>Also, kill -9 on the <code>pachctl port-forward<\/code> process 37245 doesn't work, it seems like I have to kill -9 on the <code>kubectl port-forward<\/code><\/p>",
        "Question_answer_count":2,
        "Question_comment_count":0,
        "Question_creation_date":"2017-05-30 18:42:36.22 UTC",
        "Question_favorite_count":null,
        "Question_score":4,
        "Question_tags":"kubernetes|portforwarding|pachyderm",
        "Question_view_count":438,
        "Owner_creation_date":"2012-01-22 11:03:27.597 UTC",
        "Owner_last_access_date":"2022-09-14 20:35:30.303 UTC",
        "Owner_location":"San Francisco, CA, USA",
        "Owner_reputation":3322,
        "Owner_up_votes":319,
        "Owner_down_votes":5,
        "Owner_views":160,
        "Answer_body":null,
        "Answer_comment_count":null,
        "Answer_creation_date":null,
        "Answer_last_edit_date":null,
        "Answer_score":null,
        "Question_last_edit_date":"2018-05-03 23:50:31.087 UTC",
        "Question_exclusive_tag":"Pachyderm"
    },
    {
        "Question_title":"Nodejs example for pachyderm",
        "Question_body":"<p>I am new to Pachyderm.<\/p>\n\n<p>I have a pipeline to extract, transform and then save in the db.\nEverything is already written in nodejs, docekrized.\nNow, I would like to move and use pachyderm.<\/p>\n\n<p>I tried following the python examples they provided, but creating this new pipeline always fails and the job never starts.<\/p>\n\n<p>All my code does is take the <code>\/pfs\/data<\/code> and copy it to <code>\/pfs\/out<\/code>. <\/p>\n\n<p>Here is my pipeline definition<\/p>\n\n<pre><code>{\n    \"pipeline\": {\n        \"name\": \"copy\"\n    },\n    \"transform\": {\n        \"cmd\": [\"npm\", \"start\"],\n        \"image\": \"simple-node-docker\"\n    },\n    \"input\": {\n        \"pfs\": {\n            \"repo\": \"data\",\n            \"glob\": \"\/*\"\n        }\n    }\n}\n<\/code><\/pre>\n\n<p>All that happens is that the pipeline fails and the job never starts.<\/p>\n\n<p>Is there a way to debug on why the pipeline is failing?\nIs there something special about my docker image that needs to happen?<\/p>",
        "Question_answer_count":1,
        "Question_comment_count":0,
        "Question_creation_date":"2019-05-25 20:45:13.66 UTC",
        "Question_favorite_count":null,
        "Question_score":0,
        "Question_tags":"node.js|pachyderm",
        "Question_view_count":80,
        "Owner_creation_date":"2014-06-01 13:18:45.057 UTC",
        "Owner_last_access_date":"2022-09-24 21:07:11.22 UTC",
        "Owner_location":null,
        "Owner_reputation":71,
        "Owner_up_votes":9,
        "Owner_down_votes":0,
        "Owner_views":2,
        "Answer_body":null,
        "Answer_comment_count":null,
        "Answer_creation_date":null,
        "Answer_last_edit_date":null,
        "Answer_score":null,
        "Question_last_edit_date":null,
        "Question_exclusive_tag":"Pachyderm"
    },
    {
        "Question_title":"Pachyderm pipeline does not start a job and launches an empty repo",
        "Question_body":"<p>I have a JSON configuration for my pipeline in Pachyderm:<\/p>\n\n<pre><code>{\n    \"pipeline\": {\n        \"name\": \"mopng-beneficiary-v2\"\n    },\n    \"input\": {\n        \"pfs\": {\n            \"repo\": \"mopng_beneficiary_v2\",\n            \"glob\": \"\/*\"\n        }\n    },\n    \"transform\": {\n        \"cmd\": [\"python3\", \"\/pclean_phlc9h6grzqdhm6sc0zrxjne_UdOgg.py \/pfs\/mopng_beneficiary_v2\/euoEQHIwIQTe1wXtg46fFYok.csv \/pfs\/mopng_beneficiary_v2\/\/Users\/aviralsrivastava\/Downloads\/5Feb18_master_ujjwala_latlong_dist_dno_so_v7.csv \/pfs\/mopng_beneficiary_v2\/\/Users\/aviralsrivastava\/Downloads\/ppac_master_v3_mmi_enriched_with_sanity_check.csv \/pfs\/mopng_beneficiary_v2\/Qc.csv\"],\n        \"image\": \"mopng-beneficiary-v2-image\"\n    }\n}\n<\/code><\/pre>\n\n<p>And my docker file is as follows:<\/p>\n\n<pre><code>FROM ubuntu:14.04\n\n# Install opencv and matplotlib.\nRUN apt-get update \\\n    &amp;&amp; apt-get upgrade -y \\\n    &amp;&amp; apt-get install -y unzip wget build-essential \\\n        cmake git pkg-config libswscale-dev \\\n        python3-dev python3-numpy python3-tk \\\n        libtbb2 libtbb-dev libjpeg-dev \\\n        libpng-dev libtiff-dev libjasper-dev \\\n        bpython python3-pip libfreetype6-dev \\\n    &amp;&amp; apt-get clean \\\n    &amp;&amp; rm -rf \/var\/lib\/apt\n\nRUN sudo pip3 install matplotlib\nRUN sudo pip3 install pandas\n\n# Add our own code.\nADD pclean.py \/pclean.py\n<\/code><\/pre>\n\n<p>However, when I run my command to create the pipeline:<\/p>\n\n<pre><code>pachctl create-pipeline -f https:\/\/raw.githubusercontent.com\/avisrivastava254084\/learning-pachyderm\/master\/pipeline.json\n<\/code><\/pre>\n\n<p>The files are existing in the pfs:<\/p>\n\n<pre><code>pachctl put-file mopng_beneficiary_v2 master -f \/Users\/aviralsrivastava\/Downloads\/pclean_phlc9h6grzqdhm6sc0zrxjne_UdOgg.py\n\u279c  ~ pachctl put-file mopng_beneficiary_v2 master -f \/Users\/aviralsrivastava\/Downloads\/5Feb18_master_ujjwala_latlong_dist_dno_so_v7.csv\n\u279c  ~ pachctl put-file mopng_beneficiary_v2 master -f \/Users\/aviralsrivastava\/Downloads\/ppac_master_v3_mmi_enriched_with_sanity_check.csv\n\u279c  ~ pachctl put-file mopng_beneficiary_v2 master -f \/Users\/aviralsrivastava\/Downloads\/euoEQHIwIQTe1wXtg46fFYok.csv\n<\/code><\/pre>\n\n<p>It should be worth to note that I am getting this from the logs command(<code>pachctl get-logs --pipeline=mopng-beneficiary-v2<\/code>):<\/p>\n\n<pre><code>container \"user\" in pod \"pipeline-mopng-beneficiary-v2-v1-lnbjh\" is waiting to start: trying and failing to pull image\n<\/code><\/pre>",
        "Question_answer_count":1,
        "Question_comment_count":1,
        "Question_creation_date":"2018-12-31 13:15:46.783 UTC",
        "Question_favorite_count":null,
        "Question_score":0,
        "Question_tags":"json|docker|pachyderm",
        "Question_view_count":232,
        "Owner_creation_date":"2018-07-10 15:06:30.283 UTC",
        "Owner_last_access_date":"2019-02-13 09:38:09.97 UTC",
        "Owner_location":"delhi",
        "Owner_reputation":903,
        "Owner_up_votes":53,
        "Owner_down_votes":0,
        "Owner_views":144,
        "Answer_body":null,
        "Answer_comment_count":null,
        "Answer_creation_date":null,
        "Answer_last_edit_date":null,
        "Answer_score":null,
        "Question_last_edit_date":"2019-07-15 16:18:17.823 UTC",
        "Question_exclusive_tag":"Pachyderm"
    },
    {
        "Question_title":"Pachyderm deploy GCP - no such image",
        "Question_body":"<p>I'm deploying Pachyderm on GKE but when I deploy the pipeline (following the <a href=\"https:\/\/docs.pachyderm.com\/latest\/getting_started\/beginner_tutorial\/\" rel=\"nofollow noreferrer\">https:\/\/docs.pachyderm.com\/latest\/getting_started\/beginner_tutorial\/<\/a>) the Pod fails in ImagePullCrashLoopBack giving this error &quot;no such image&quot;.<\/p>\n<p>Here, the output of the command &quot;kubectl get pods&quot;:\n<a href=\"https:\/\/i.stack.imgur.com\/lpvj7.png\" rel=\"nofollow noreferrer\">screenshot<\/a><\/p>\n<p>How can I fix the deployment procedure?<\/p>",
        "Question_answer_count":1,
        "Question_comment_count":1,
        "Question_creation_date":"2021-06-11 13:26:09.943 UTC",
        "Question_favorite_count":null,
        "Question_score":0,
        "Question_tags":"kubernetes|google-cloud-platform|pachyderm",
        "Question_view_count":48,
        "Owner_creation_date":"2014-11-06 09:41:52.943 UTC",
        "Owner_last_access_date":"2022-09-24 19:13:38.297 UTC",
        "Owner_location":"Milano, Metropolitan City of Milan, Italy",
        "Owner_reputation":107,
        "Owner_up_votes":27,
        "Owner_down_votes":0,
        "Owner_views":24,
        "Answer_body":"<p>As mentioned in the Slack channel of Pachyderm community, adding the flag <code>--no-expose-docker-socket<\/code> to the deploy call should solve the issue.<\/p>\n<p><code>pachctl deploy google ${BUCKET_NAME} ${STORAGE_SIZE} --dynamic-etcd-nodes=1 --no-expose-docker-socket<\/code><\/p>",
        "Answer_comment_count":1.0,
        "Answer_creation_date":"2021-06-11 13:29:00.677 UTC",
        "Answer_last_edit_date":null,
        "Answer_score":1.0,
        "Question_last_edit_date":null,
        "Question_exclusive_tag":"Pachyderm"
    }
]